src/relayserver/RelayServer.ts

```diff
@@ -594,50 +614,97 @@ export class RelayServer extends EventEmitter {
       // register on chain
       const addRelayWorkerMethod = this.relayHubContract?.contract.methods
         .addRelayWorkers([this.getAddress(1)])
-      await this._sendTransaction({
+      receipts = receipts.concat((await this._sendTransaction({
         signerIndex: 0,
         method: addRelayWorkerMethod,
         destination: this.relayHubContract?.address as string
-      })
+      })).receipt)
     }
-    const registerMethod = this.relayHubContract?.contract.methods
-      .registerRelayServer(this.baseRelayFee, this.pctRelayFee,
-        this.url)
-    const { receipt } = await this._sendTransaction({
-      signerIndex: 0,
-      method: registerMethod,
-      destination: this.relayHubContract?.address as string
+    const relayRegisteredEvents = await this.relayHubContract?.contract.getPastEvents('RelayServerRegistered', {
+      fromBlock: 1,
+      filter: { relayManager: this.managerAddress }
     })
-    debug(`Relay ${this.managerAddress} registered on hub ${this.relayHubContract?.address}. `)
-
+    if (relayRegisteredEvents.find(
+      (e: any) =>
+        e.returnValues.relayManager.toLowerCase() === this.managerAddress.toLowerCase() &&
+        e.returnValues.baseRelayFee.toString() === this.baseRelayFee.toString() &&
+        e.returnValues.pctRelayFee.toString() === this.pctRelayFee.toString() &&
+        e.returnValues.relayUrl.toString() === this.url.toString()) == null) {
```

Comment:
maybe create something (function, lambda, idk) for this, it is a long ```find```

Reply:
```
      (e: any) =>
        e.returnValues.relayManager.toLowerCase() === this.managerAddress.toLowerCase() &&
        e.returnValues.baseRelayFee.toString() === this.baseRelayFee.toString() &&
        e.returnValues.pctRelayFee.toString() === this.pctRelayFee.toString() &&
        e.returnValues.relayUrl.toString() === this.url.toString()
```
Isn't it obvious that it compares all of the event entries to see if it was already registered with the same params? It seems pretty obvious to me